crates/shared/src/price_estimation/instrumented.rs

```diff
@@ -42,7 +30,20 @@ impl PriceEstimating for InstrumentedPriceEstimator {
         &'a self,
         queries: &'a [Query],
     ) -> futures::stream::BoxStream<'_, (usize, super::PriceEstimateResult)> {
-        old_estimator_to_stream(self.estimates_(queries))
+        let start = Instant::now();
+        let measure_time = async move {
+            self.metrics
+                .price_estimation_timed(&self.name, start.elapsed());
+        };
+        self.inner
+            .estimates(queries)
+            .map(move |result| {
+                let success = !matches!(&result.1, Err(PriceEstimationError::Other(_)));
+                self.metrics.price_estimated(&self.name, success);
+                result
+            })
+            .chain(futures::stream::once(measure_time).filter_map(|_| async { None }))
```

Comment:
I think this does the right thing but working with streams certainly takes some time getting used to. 😅